### DISCUSSION TITLE
Functions running on wrong node version

### DISCUSSION THREAD

**Discussion by maxchangm** [vercel/community#2316] [Help] [answered]
### Summary

Anyone know why despite setting Node.js version to 16.x in project settings, the functions are still running in 18.x?

![Screenshot 2023-05-05 174231](https://user-images.githubusercontent.com/19217150/236427731-d0b3039f-d5ac-41f9-bcfe-1c151649763e.png)
![Screenshot 2023-05-05 174416](https://user-images.githubusercontent.com/19217150/236427733-72114406-e12a-4c74-83d7-0b9a925fd7ba.png)

**Comment by amyegan** [comment, marked as answer]
Do you have additional configuration in your code that may be overriding the dashboard node.js version setting? The most common would be an [`engines.node`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) property in `package.json`. The build log should offer some more information about any node version overrides.

**Comment by DingoEatingFuzz** [comment]
I don't get this. If I specify a version of Node for Vercel to use that is compatible with my engines string, why would Vercel ignore that and use the latest matching version? 

Why even have the setting if you are just gonna read from engines anyway?

I would expect my Vercel setting (more specific) to override the engine string (more generic), not the other way around.

**Reply by smaeda-ks** [reply]
@DingoEatingFuzz I hear you. But in Vercel, generally, anything you define in your "code" wins and takes precedence over project settings. So `package.json` is one of them. This allows you to better track changes in your version control over time.

**Reply by DingoEatingFuzz** [reply]
Yeah, and I ❤️ that about Vercel, but I think this is a weird one. `vercel.json` taking precedence over project settings, totally get it, but `package.json`? Feels off to me.
